backend/internal/certificate.js

```diff
@@ -335,6 +336,50 @@ const internalCertificate = {
 			});
 	},
 
+	/**
+	 * @param   {Object}  data
+	 * @param   {Number}  data.id
+	 * @returns {Promise}
+	 */
+	download: (data) => {
+		const downloadName = 'npm-' + data.id + '-' + `${Date.now()}.zip`;
```

Comment:
This might work well and fine for LE certs, but this doesn't handle custom certificates. You can either return a error for them, or go to the effort of finding them and zipping them in the same way.